packages/labs/cli/src/lib/generate/react.ts

```diff
@@ -4,16 +4,188 @@
+  return JSON.stringify(
+    {
+      name: packageNameToReactPackageName(pkgJson.name!),
+      type: 'module',
+      scripts: {
+        build: 'tsc',
+        'build:watch': 'tsc --watch',
+      },
+      // TODO(kschaaf): Version in lock-step with source?
+      version: pkgJson.version,
+      dependencies: {
+        // TODO(kschaaf): make component version range configurable?
+        [pkgJson.name!]: '^' + pkgJson.version!,
+        // TODO(kschaaf): make @lit-labs/react version configurable?
+        '@lit-labs/react': '^1.0.4',
+      },
+      peerDependencies: {
+        // TODO(kschaaf): make react version(s) configurable?
+        react: '^17.0.2',
+        '@types/react': '^17.0.19',
+      },
+      devDependencies: {
+        // TODO(kschaaf): make configurable?
+        typescript: '^4.3.5',
+      },
+      files: [...litModules.map(({module}) => module.jsPath)],
+    },
+    null,
+    2
+  );
+};
+
+const gitIgnoreTemplate = (litModules: LitModule[]) => {
+  return litModules.map(({module}) => module.jsPath).join('\n');
+};
+
+const tsconfigTemplate = () => {
+  return JSON.stringify(
+    {
+      compilerOptions: {
+        target: 'es2019',
+        module: 'es2015',
+        lib: ['es2020', 'DOM', 'DOM.Iterable'],
+        declaration: true,
+        declarationMap: true,
+        sourceMap: true,
+        inlineSources: true,
+        outDir: './',
+        rootDir: './src',
+        strict: true,
+        noUnusedLocals: true,
+        noUnusedParameters: true,
+        noImplicitReturns: true,
+        noFallthroughCasesInSwitch: true,
+        noImplicitAny: true,
+        noImplicitThis: true,
+        moduleResolution: 'node',
+        allowSyntheticDefaultImports: true,
+        experimentalDecorators: true,
+        noImplicitOverride: true,
+      },
+      include: ['src/**/*.ts'],
+      exclude: [],
+    },
+    null,
+    2
   );
 };
+
+const wrapperModuleTemplate = (
+  packageJson: PackageJson,
+  moduleJsPath: string,
+  elements: LitElementDeclaration[]
+) => {
+  return javascript`
+import * as React from 'react';
+import {createComponent} from '@lit-labs/react';
+${elements.map(
+  (element) => javascript`
+import {${element.name} as ${element.name}Element} from '${packageJson.name}/${moduleJsPath}';
+`
+)}
+
+${elements.map((element) => wrapperTemplate(element))}
+`;
+};
+
+// TODO(kschaaf): Should this be configurable?
+const packageNameToReactPackageName = (pkgName: string) => `${pkgName}-react`;
+
+const eventNameToCallbackName = (eventName: string) =>
+  'on' +
+  eventName[0].toUpperCase() +
+  eventName.slice(1).replace(/-[a-z]/g, (m) => m[1].toUpperCase());
+
+const wrapperTemplate = ({name, tagname, events}: LitElementDeclaration) => {
+  return javascript`
+export const ${name} = createComponent(
+  React,
+  '${tagname}',
+  ${name}Element,
+  {
+    ${Array.from(events.keys()).map(
+      (eventName) => javascript`
+    ${eventNameToCallbackName(eventName)}: '${eventName}',`
```

Comment:
We starting adding ` as EventName<MouseEvent>` for typing. If we don't have metadata to add this yet, maybe just make an issue?

Reply:
Added issue/todo. https://github.com/lit/lit/issues/2850